### PR TITLE
Enhancement/130

### PR DESCRIPTION
**Pull request by PeteSwauger**
Performance timing added
Typescript 3

### CODE REVIEW

File: Jenkinsfile [hunk 1]
```diff
@@ -9,163 +9,896 @@
 *                                                                                 *
 */
 
-@Library('shared-pipelines@v1.0.2') import org.zowe.pipelines.nodejs.NodeJSPipeline
+// The following property need to be set for the HTML report @TODO figure out how to get this nicely on jenkins
+//System.setProperty("hudson.model.DirectoryBrowserSupport.CSP", "")
```
Comment (ChrisBoehmCA) [inline Jenkinsfile]: It kind of looks like you accidentally reverted the Jenkinsfile back to the old one, right @zFernand0 ?

Comment (ChrisBoehmCA) [reply Jenkinsfile]: Fixed

File: packages/main.ts [hunk 2]
```diff
@@ -10,6 +10,11 @@
 *
 */
 
+import { PerfTiming } from "@zowe/perf-timing";
+
+const timingApi = PerfTiming.api;
+
+timingApi.mark("PRE-INIT");
```
Comment (ChrisBoehmCA) [inline packages/main.ts]: Not a big deal, but I wonder if we could  change the name as "PRE-INIT" vs. "BEFORE_INIT" is a little confusing to me. Maybe "PRE_IMPORT_IMPERATIVE" and "POST_IMPORT_IMPERATIVE" or something like that?

Also even if we don't change the name, I think you have a hyphen here but an underscore in the call to .measure below so the measurement might not work without an update

Comment (PeteSwauger) [reply packages/main.ts]: I am okay with any change.  The only thing I wanted was all uppercase so they stood out.

Comment (zFernand0) [reply packages/main.ts]: I'm so sorry I missed this 😢

**Pull request merged**